src/plugins/vis_type_timelion/public/components/timelion_vis_component.tsx

```diff
@@ -85,279 +52,77 @@ function TimelionVisComponent({
       const newSeries = { ...series };
       if (!newSeries.color) {
         const colorIndex = seriesIndex % colors.length;
         newSeries.color = colors[colorIndex];
       }
-      colorsSet.push([newSeries, newSeries.color]);
-      return newSeries;
-    });
-    setChart(newChart);
-    setOriginalColorMap(new Map(colorsSet));
-  }, [seriesList.list]);
-
-  const unhighlightSeries = useCallback(() => {
-    if (highlightedSeries === null) {
-      return;
-    }
-
-    setHighlightedSeries(null);
-    setFocusedSeries(null);
-
-    setChart((chartState) =>
-      chartState.map((series: Series) => {
-        series.color = originalColorMap.get(series); // reset the colors
-        return series;
-      })
-    );
-  }, [originalColorMap, highlightedSeries]);
-
-  // Shamelessly borrowed from the flotCrosshairs example
-  const setLegendNumbers = useCallback(
-    (pos: Position) => {
-      unhighlightSeries();
-
-      const axes = plot!.getAxes();
-      if (pos.x < axes.xaxis.min! || pos.x > axes.xaxis.max!) {
-        return;
-      }
 
-      const dataset = plot!.getData();
-      if (legendCaption) {
-        legendCaption.text(
-          moment(pos.x).format(get(dataset, '[0]._global.legend.timeFormat', DEFAULT_TIME_FORMAT))
-        );
+      if (newSeries._global?.yaxes) {
+        updateYAxes(newSeries._global.yaxes);
       }
-      for (let i = 0; i < dataset.length; ++i) {
-        const series = dataset[i];
-        const useNearestPoint = series.lines!.show && !series.lines!.steps;
-        const precision = get(series, '_meta.precision', 2);
-
-        // We're setting this flag on top on the series object belonging to the flot library, so we're simply casting here.
-        if ((series as { _hide?: boolean })._hide) {
-          continue;
-        }
 
-        const currentPoint = series.data.find((point: [number, number], index: number) => {
-          if (index + 1 === series.data.length) {
-            return true;
-          }
-          if (useNearestPoint) {
-            return pos.x - point[0] < series.data[index + 1][0] - pos.x;
-          } else {
-            return pos.x < series.data[index + 1][0];
-          }
-        });
+      return newSeries;
+    });
 
-        const y = currentPoint[1];
+    setChart(newChart);
+  }, [seriesList.list]);
 
-        if (legendValueNumbers) {
-          if (y == null) {
-            legendValueNumbers.eq(i).empty();
-          } else {
-            let label = y.toFixed(precision);
-            const formatter = ((series.yaxis as unknown) as Axis).tickFormatter;
-            if (formatter) {
-              label = formatter(Number(label), (series.yaxis as unknown) as Axis);
-            }
-            legendValueNumbers.eq(i).text(`(${label})`);
-          }
-        }
+  useEffect(() => {
+    setOptions(buildOptions(interval, kibana.services.timefilter, kibana.services.uiSettings));
+  }, [interval, kibana.services.timefilter, kibana.services.uiSettings]);
```

Comment:
I'm not sure why do we need such a complex design here with `useState` & `useEffect` here.
I suppose those could be simply replaced with:
```suggestion
  const options = useMemo(() => buildOptions(interval, services), [interval, services]);
```

but also since I haven't found usages of options except the `options.xaxis.tickFormatter`, this could be simplified more:

```suggestion
  const tickFormat = useMemo(() => createTickFormat(interval, services), [interval, services]);
```

Reply:
done